src/plasticScm.ts

```diff
@@ -1,12 +1,16 @@
 
 import { workspace, Disposable, OutputChannel } from 'vscode';
-import { ICmShell, CmShell } from './cmShell';
+import { ICmShell, CmShell, ICmdResult } from './cmShell';
 import * as os from 'os';
+import { GetWorkspaceFromPathResult, GetWorkspaceFromPath } from './commands/getWorkspaceFromPath';
+import * as vscode from 'vscode';
 
 class Workspace implements Disposable {
-  constructor(path: string, name: string, shell: ICmShell) {
+
+  constructor(path: string, name: string, id: string, shell: ICmShell) {
```

Comment:
This `Workspace` class should receive on its constructor a `WorkspaceInfo` object (see previous comments), and the `ICmShell` that was initialized earlier.